### PR TITLE
Update projects to include App.framework for release mode

### PR DESCRIPTION
**Pull request by jonahwilliams**
Requires https://github.com/flutter/flutter/pull/37425

Adds an App.framework and places assets under this framework. This project can be transparently updated to support release/profile once the artifacts are available.

### CODE REVIEW

File: example/macos/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -389,9 +412,13 @@
 				FRAMEWORK_SEARCH_PATHS = (
 					$PROJECT_DIR/Flutter/ephemeral,
 					"$(inherited)",
+					"$(PROJECT_DIR)/Flutter/ephemeral",
```
Comment (stuartmorgan) [inline example/macos/Runner.xcodeproj/project.pbxproj]: Can you fix this duplication of paths? If this was done via the Xcode UI, then remove the one a couple of lines up.

File: example/macos/Runner.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -411,9 +438,13 @@
...
```
Comment (stuartmorgan) [inline example/macos/Runner.xcodeproj/project.pbxproj]: Same.

File: testbed/macos/Runner.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -453,11 +466,15 @@
 				COMBINE_HIDPI_IMAGES = YES;
 				DEVELOPMENT_TEAM = "";
 				FRAMEWORK_SEARCH_PATHS = (
-					$PROJECT_DIR/Flutter/ephemeral,
+					"$PROJECT_DIR/Flutter/ephemeral",
 					"$(inherited)",
+					"$(PROJECT_DIR)/Flutter/ephemeral",
```
Comment (stuartmorgan) [inline testbed/macos/Runner.xcodeproj/project.pbxproj]: Same.

File: testbed/macos/Runner.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -476,11 +493,15 @@
...
```
Comment (stuartmorgan) [inline testbed/macos/Runner.xcodeproj/project.pbxproj]: Same.

**Comment by jonahwilliams** [comment]
Removed extra search paths and updated to latest desktop

**Pull request merged**